rqscheduler.py

```diff
@@ -4,10 +4,33 @@
 # implementation. This is required for correct work with CVAT queue settings and
 # their access options such as login and password.
 
+import argparse
+import os
+import sys
+
 # Required to initialize Django settings correctly
 from cvat.asgi import application  # pylint: disable=unused-import
 
+from redis import Redis
+from rq import Queue
 from rq_scheduler.scripts import rqscheduler
 
 if __name__ == '__main__':
-    rqscheduler.main()
+    parser = argparse.ArgumentParser(add_help=False)
+    parser.add_argument('-H', '--host', default=os.environ.get('RQ_REDIS_HOST', 'localhost'), help="Redis host")
+    parser.add_argument('-p', '--port', default=int(os.environ.get('RQ_REDIS_PORT', 6379)), type=int, help="Redis port number")
+    parser.add_argument('-d', '--db', default=int(os.environ.get('RQ_REDIS_DB', 0)), type=int, help="Redis database")
+    parser.add_argument('-P', '--password', default=os.environ.get('RQ_REDIS_PASSWORD'), help="Redis password")
+    parser.add_argument('--clear', action='store_true', help="Clean up all jobs from all queues")
+    args, _ = parser.parse_known_args()
+
+    if args.clear:
+        connection = Redis(args.host, args.port, args.db, args.password)
+        for queue in Queue.all(connection=connection):
+            for job in queue.jobs:
+                job.cancel()
+
+        sys.exit(0)
```

Comment:
I don't think creating an extra script just for this is a good idea. There is the existing `burst` parameter, which is quite similar in behavior, so I decided to put this queue management code here.

Reply:
> They are not similar in behavior. rqscheduler works with scheduled tasks, whereas your code works with queued tasks.

Generally, this is what schedulers do - they manage queues and task order, the specific queue is a detail.

> But besides that, it just doesn't fit the description of the script.

Do you mean the comment on top of the file? Ok, I'll update the description.

> (Incidentally, I'm not sure why we even have this script at all. Seems like it could just be replaced with manage.py rqscheduler.)

This is in the description. Without the application import line, the original implementation cannot work with queue parameters from the django application config. The `manage.py rqscheduler` has different parameters from the original `rqscheduler.py`, and these parameters don't allow us to have 1 scheduler for all queues.

Reply:
> Generally, this is what schedulers do - they manage queues and task order, the specific queue is a detail.

I'm not talking generally, I'm talking specifically about RQ Scheduler. The purpose of `rqscheduler.main()` is to find scheduled tasks and enqueue them. Whereas your code finds _enqueued_ tasks (which might not have even come from the scheduler) and deletes them. There's no relation.

Your code doesn't even use the `rq_scheduler` module. That seems to me like a sign that it shouldn't be in `rqscheduler.py`.

> The manage.py rqscheduler has different parameters from the original rqscheduler.py, and these parameters don't allow us to have 1 scheduler for all queues.

I see... that's pretty annoying.

Frankly, it seems like a goof by django-rq, because I don't think `Scheduler.run` even uses the queue that was passed in in the constructor. It processes all scheduled jobs regardless of queue. I think I'll file an issue for django-rq.

Reply:
> Frankly, it seems like a goof by django-rq, because I don't think Scheduler.run even uses the queue that was passed in in the constructor. It processes all scheduled jobs regardless of queue. I think I'll file an issue for django-rq.

Oh, I see why django-rq does this. It's because it allows you to configure different servers per queue (rq/django-rq#28), so you need to know the name of the queue to connect to the appropriate server. Less of a goof and more of a design error.

Except... hold on. If you need to know the name of the queue to connect to the appropriate server, then how could this script have worked in the first place? How would merely importing `cvat.asgi` help it connect?